src/freertos_drivers/common/WiFiInterface.hxx

```diff
@@ -0,0 +1,377 @@
+    /// is only valid for @ref WiFiSecurity::OPEN.
+    /// @param security @ref WiFiSecurity of the profile to be saved.
+    /// @param priority connection priority when more than one of the saved
+    /// networks is available, 0 == lowest priority
+    /// @param profile_index will contain the index of the newly stored
+    /// profile, if successfully stored.
+    ///
+    /// @return @ref WiFiErrorCode as the result of storing the profile.
+    virtual WiFiErrorCode add_ssid_profile(const std::string &ssid,
+                                           const std::string &password,
+                                           const WiFiSecurity security,
+                                           const uint8_t priority,
+                                           uint8_t *profile_index) = 0;
+
+    /// Delete a saved SSID profile by it's SSID.
+    ///
+    /// @param ssid SSID of the profile to delete.
+    ///
+    /// @return @ref WiFiErrorCode as the result of the removal of the profile.
+    virtual WiFiErrorCode del_ssid_profile(const std::string &ssid) = 0;
+
+    /// Delete a saved SSID profile by it's index.
+    ///
+    /// @param index index within saved profile list to remove, a value of 0xFF
+    /// will remove all stored profiles.
+    ///
+    /// @return @ref WiFiErrorCode as the result of the removal of the profile.
+    virtual WiFiErrorCode del_ssid_profile(const int index) = 0;
+
+    /// Retrieves a saved SSID profile by index.
+    ///
+    /// @param index of the stored profile to retrieve.
+    /// @param ssid will contain the ssid of the profile.
+    /// @param security will contain the security type of the profile.
+    /// @param priority will contain the priority of the profile.
+    ///
+    /// @return @ref WiFiErrorCode as the result of the retrieval of the
+    /// profile.
+    virtual WiFiErrorCode get_ssid_profile(const int index, std::string *ssid,
+                                           WiFiSecurity *security,
+                                           uint8_t *priority) = 0;
+
+    /// @return true if there are no saved SSID profiles, false otherwise.
+    virtual bool are_ssid_profiles_empty() = 0;
+
+    /// Initiate scanning for available access points.
+    ///
+    /// @param blocking will cause this method to block until the network scan
+    /// completes. Default is non-blocking.
+    /// @param timeout is the number of milliseconds to allow for the network
+    /// scan to complete. This is only valid when @ref blocking is true.
+    /// @param max_results is the maximum number of results to scan for. If the
+    /// WiFi interface supports a lower number than requested and in such case
+    /// the WiFi interface supported limit will be used instead.
+    ///
+    /// @return the number of networks found via scan, this value will be zero
+    /// when @ref blocking is false.
+    virtual int initiate_network_scan(bool blocking = false,
```

Comment:
The CC32xx supports both one-shot and continuous scan.  On platforms that do not support continuous scan, it can be emulated with a periodic one-shot scan.  Should we add an additional API for continuous scan enable/disable?

Reply:
I'll add a second set of APIs for continuous.